File: benchmarks/python/url.py
from urllib.parse import parse_qs, quote, unquote, urlparse


def url_score(n):
    i = 0
    acc = 0
    target = "https://agent.local/tools/search?q=Ax%20language&mode=fast"
    while i < n:
        parsed = urlparse(target)
        query = parse_qs(parsed.query).get("q", [""])[0]
        encoded = quote(parsed.path, safe="")
        decoded = unquote(encoded)
        acc += len(parsed.hostname or "")
        acc += len(decoded)
        acc += len(query)
        i += 1
    return acc % 251


raise SystemExit(url_score(100_000))

File: benchmarks/js/url.js
function urlScore(n) {
  let i = 0;
  let acc = 0;
  const target = "https://agent.local/tools/search?q=Ax%20language&mode=fast";
  while (i < n) {
    const parsed = new URL(target);
    const query = parsed.searchParams.get("q") ?? "";
    const encoded = encodeURIComponent(parsed.pathname);
    const decoded = decodeURIComponent(encoded);
    acc += parsed.hostname.length;
    acc += decoded.length;
    acc += query.length;
    i += 1;
  }
  return acc % 251;
}

process.exitCode = urlScore(100000);
